CD pipeline that deploys resources to GKE cluster

KubaKrzych opened the pull request:
The product of this branch is a completed CD pipeline that deploys resources to GKE. 

The mentioned resources are Docker images stored in GCR and/or images stored in Docker hub.

I'll be grateful for any tips and comments.

matined commented on k8s/values-backend.yaml:
```diff
@@ -0,0 +1,37 @@
+    - name: POSTGRES_HOST
+      value: "database-service"
+    - name: POSTGRES_PASSWORD
+      value: "postgres"
```
we need to store this value in secrets, I created POSTGRES_PASSWORD

matined commented on k8s/values-backend.yaml:
```diff
@@ -0,0 +1,37 @@
+deployment:
+  replicas: 1
+  name: api
```
let's stay consistent and use backend instead of api everywhere

matined commented on k8s/values-database.yaml:
```diff
@@ -0,0 +1,37 @@
+      value: "postgres"
+    - name: POSTGRES_PASSWORD
+      value: "postgres"
+    - name: POSTGRES_DB
```
I created two more secrets: POSTGRES_USER, POSTGRES_DATABASE

matined commented on k8s/values-backend.yaml:
```diff
@@ -0,0 +1,37 @@
+  template:
+    spec:
+      container:
+        image: gcr.io/${{ secrets.GOOGLE_PROJECT_ID }}/backend
```
I wonder if it works

KubaKrzych replied:
Well, I'm positive that it doesn't work. Do you think a better option would be to pass a project_id as an environmental variable?

I'm if the GCP project ID is a vulnerable data, if not then I can pass the project ID explicitily.

matined commented:
One more request is to rename `k8s` folder to `kubernetes`

KubaKrzych commented:
I'm pretty sure I've added and changed everything you've mentioned. 

I've decided to stick with the deployment of backend/front/... in a way it was before. I've left untouched those if-statements that we've chatted about. In the long term I think we should think about something else, but it's cool for now.
